src/utils/avatar.js

```diff
@@ -389,7 +389,7 @@ export class UploadedAvatarURL extends AvatarURL {
          */
       result = new URL(this._standardUrl);
 
-      const match = new RegExp(/\.png$/g).exec(result.pathname);
```

Comment:
UploadedAvatarURL: Remove global flag in regex.
    
    Since the regex matches end-of-line, there should be at most one
    match.

Yeah. Also even if there were several matches (like with the original pattern), we only call this `RegExp` object once before discarding it -- so it'll return the first match then anyway, and we never go back to ask it about a next match.

Quick demo in a browser console:
```
> r = /(\w+)\.png/g
/(\w+)\.png/g

> r.exec("foo.png/bar.png")
(2) ["foo.png", "foo", index: 0, input: "foo.png/bar.png", groups: undefined]

> r.exec("foo.png/bar.png")
(2) ["bar.png", "bar", index: 8, input: "foo.png/bar.png", groups: undefined]

> r.exec("foo.png/bar.png")
null
```

Reply:
To be clear, I think the existing pattern's behavior isn't a *live* bug, only a latent one -- I believe Zulip avatar URLs all follow highly structured shapes where the only place the pathname can match this is at the end. But certainly good to fix! As well as to simplify the logic, and to fix the original motivating issue here.

Reply:
Yeah—I spotted that we could remove the `g` because of its different behavior when used as a pattern in `.replace`. IIRC/IIUC it was trying to replace multiple things at once; I didn't investigate super closely after realizing it should just go away in any case.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace
> To perform a global search and replace, include the g switch in the regular expression.